core/mobile/src/backend/MobileAuthorizationBackend.ts

```diff
@@ -69,12 +113,53 @@ export class MobileAuthorizationBackend extends NativeAppAuthorizationBackend {
     });
   }
 
+  public setAccessToken(token?: AccessToken) {
+    if (token === this._accessToken)
+      return;
+    this._accessToken = token;
+  }
+
+  public async getAccessToken(): Promise<AccessToken> {
+    if (!this._accessToken)
+      this.setAccessToken(await this.refreshToken());
+    return this._accessToken ?? "";
+  }
+
   /** return accessToken */
   public async refreshToken(): Promise<AccessToken> {
-    return new Promise<AccessToken>((resolve) => {
-      MobileHost.device.authGetAccessToken((tokenStringJson?: AccessToken) => {
-        resolve(tokenStringJson ?? "");
+    return new Promise<AccessToken>((resolve, reject) => {
+      MobileHost.device.authGetAccessToken((tokenStringJson?: AccessToken, err?: string) => {
+        if (!err && tokenStringJson) {
+          resolve(tokenStringJson);
+        } else {
+          reject(new Error(err));
+        }
       });
     });
   }
+  /**
+   * Gets the URL of the service. Uses the default URL provided by client implementations.
+   * If defined, the value of `IMJS_URL_PREFIX` will be used as a prefix to all urls provided
+   * by the client implementations.
+   *
+   * Note that for consistency sake, the URL is stripped of any trailing "/".
+   * @returns URL for the service
+   */
+  public getUrl(): string {
```

Comment:
What's the reason to make this public?

Reply:
I think this was a product of copy/pasting from "@bentley/itwin-client" client class when flattening the hierarchy of this class. It is public there. I can make it private if you don't foresee us ever needing to publicly get the URL from this class.

Reply:
I don't think it's necessary in this case for the mobile authorization at least. Let's make it private for now.

Reply:
Should be fixed now.